src/main/java/de/tum/in/www1/artemis/repository/StudentExamRepository.java

```diff
@@ -374,7 +374,15 @@ default Integer findMaxWorkingTimeByExamIdElseThrow(Long examId) {
     default List<StudentExam> createRandomStudentExams(Exam exam, Set<User> users, ExamQuizQuestionsGenerator examQuizQuestionsGenerator) {
         List<StudentExam> studentExams = new ArrayList<>();
         SecureRandom random = new SecureRandom();
-        long numberOfOptionalExercises = exam.getNumberOfExercisesInExam() - exam.getExerciseGroups().stream().filter(ExerciseGroup::getIsMandatory).count();
+        long numberOfOptionalExercises = 0;
+
+        // In case the total number of exercises in the exam is not set by the instructor
+        try {
+            numberOfOptionalExercises = exam.getNumberOfExercisesInExam() - exam.getExerciseGroups().stream().filter(ExerciseGroup::getIsMandatory).count();
+        }
+        catch (NullPointerException e) {
+            throw new EntityNotFoundException("The number of exercises in the exam " + exam.getId() + " does not exist");
+        }
```

Comment:
Please don't catch NPEs, but check for null values (assuming this method returns a boxed Integer)
```suggestion
       if (exam.getNumberOfExercisesInExam() != null) {
            numberOfOptionalExercises = exam.getNumberOfExercisesInExam() - exam.getExerciseGroups().stream().filter(ExerciseGroup::getIsMandatory).count();
        }
        else {
            throw new EntityNotFoundException("The number of exercises in the exam " + exam.getId() + " does not exist");
        }
```